.github/workflows/build.yml

```diff
@@ -54,12 +55,18 @@ jobs:
           echo -e 'package blaze-textual\n    flags: +integer-simple' >> cabal.project.local
         fi
 
-    - name: (GHC 9.0) Use modified cabal.project for GHC 9.0
-      if: ${{ matrix.ghc == '9.0.1' }}
-      run: cp cabal-ghc901.project cabal.project
-    - name: Use modified cabal.project for ghc9.2
-      if: ${{ matrix.ghc == '9.2.1' }}
-      run: cp cabal-ghc921.project cabal.project
+    - name: Use modified cabal.project
```

Comment:
Wait, shouldn't this just be using the setup-build workflow from above?

Reply:
yeah, it should but we were not using caching here, not sure why. I had that as TODO in my mind :-P

Reply:
my idea is check if caching should be effectively avoided and separate the actual action in two, extracting the cache part, in the other case

Reply:
Yes. Neither of us initially planne to refactor `build`. `build` seems like particularly rarely used & important workflow & I left it to the more experienced people. Consolidating & solidifying the work between `caching`, `test`, `bench` already was enough task & scope. We always can share the work done into `build`, but at the start that approach was ineffective (consolidating between 4 targets at once may be exponentially harder task).

Reply:
Reported in https://github.com/haskell/haskell-language-server/issues/2586